### PR TITLE
feat: get files for a session, skip files with wrong extension in uploads

### PR DESCRIPTION
**Pull request by rjanjua**
Makes understanding why uploads failed simplre

### Checklist

- [ ] I have updated the referenced issue with new insights and changes
- [ ] If this is a code change, I have added unit tests
- [ ] I've used the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
- [ ] I updated the docstrings
- [ ] If this is a code change, I added meaningful logs and prepared Datadog visualizations and alerts

### CODE REVIEW

File: deepset_cloud_sdk/_service/files_service.py [hunk 1]
```diff
@@ -324,10 +329,21 @@ async def upload_texts(
...
+
+        files_that_can_be_uploaded = []
+        allowed_suffixes = {".txt", ".json", ".pdf"}
+
+        for file in files:
+            file_as_path = Path(file.name)
+            if file_as_path.suffix.lower() in allowed_suffixes:
+                files_that_can_be_uploaded.append(file)
+            else:
+                logger.warning("Skipping file as file extension not allowed, must be .txt or .pdf", filename=file.name)
```
Comment (mathislucka) [inline deepset_cloud_sdk/_service/files_service.py:341]: or .json?

File: deepset_cloud_sdk/_api/upload_sessions.py [hunk 2]
```diff
@@ -275,3 +316,62 @@ async def list(
...
+    )
+    async def list_session_files(
+        self,
+        workspace_name: str,
+        session_id: UUID,
+        ingestion_status: FileIngestionStatus,
+        limit: int = 10,
+        page_number: int = 1,
+    ) -> UploadSessionFileList:
+        """List upload sessions.
```
Comment (ArzelaAscoIi) [inline deepset_cloud_sdk/_api/upload_sessions.py:334]: I guess copy&paste error ?

File: deepset_cloud_sdk/_service/files_service.py [hunk 3]
```diff
@@ -324,10 +329,21 @@ async def upload_texts(
         :show_progress If True, shows a progress bar for S3 uploads.
         :raises TimeoutError: If blocking is True and the ingestion takes longer than timeout_s.
         """
+
+        files_that_can_be_uploaded = []
+        allowed_suffixes = {".txt", ".json", ".pdf"}
```
Comment (ArzelaAscoIi) [inline deepset_cloud_sdk/_service/files_service.py:334]: We could move this to ca constant

Comment (ArzelaAscoIi) [reply deepset_cloud_sdk/_service/files_service.py:334]: (and unify this a bit )

**Review by ArzelaAscoIi — COMMENTED** [review]
Amazing! Thanks do pushing this. Fetching the files of a session will be super valuable. 
Lets try to covers some more lines here if it doesn't take too long. 

I guess you already though about it. If possible it would be super nice to split the Bugfix/feature into separate PRs.

**Comment by ArzelaAscoIi** [comment]
Stale: closing

**Pull request closed without merging**